File: src/Entity/Categorie.php
<?php
// src/Entity/Categorie.php

namespace App\Entity;

use App\Repository\CategorieRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: CategorieRepository::class)]
class Categorie
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)] 
    #[Assert\NotBlank(message: "Le nom de catégorie est obligatoire.")]
    #[Assert\Length(
        min: 3,
        max: 25,
        minMessage: "Le nom de catégorie doit contenir au moins {{ limit }} caractères.",
        maxMessage: "Le nom de catégorie ne peut pas dépasser {{ limit }} caractères."
    )]
    private ?string $nomCategorie = null;

    #[ORM\OneToMany(targetEntity: Cours::class, mappedBy: 'categorie', cascade: ['remove'])]
    private Collection $cours;

    public function __construct()
    {
        $this->cours = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomCategorie(): ?string
    {
        return $this->nomCategorie;
    }

    public function setNomCategorie(string $nomCategorie): static
    {
        $this->nomCategorie = $nomCategorie;

        return $this;
    }

    public function getCours(): Collection
    {
        return $this->cours;
    }

    public function addCour(Cours $cour): static
    {
        if (!$this->cours->contains($cour)) {
            $this->cours->add($cour);
            $cour->setCategorie($this);
        }

        return $this;
    }

    public function removeCour(Cours $cour): static
    {
        if ($this->cours->removeElement($cour)) {
            if ($cour->getCategorie() === $this) {
                $cour->setCategorie(null);
            }
        }

        return $this;
    }

    public function __toString(): string
    {
        return $this->nomCategorie ?? 'N/A';
    }
}

File: Educonnect/src/main/java/com/esprit/Models/Categorie.java
package com.esprit.Models;

import java.util.ArrayList;
import java.util.List;

public class Categorie {
    private int id;
    private String nomCategorie;
    private List<Cours> cours = new ArrayList<>();

    // Constructors
    public Categorie() {}

    public Categorie(String nomCategorie) {
        this.nomCategorie = nomCategorie;
    }

    public Categorie(int id, String nomCategorie) {
        this.id = id;
        this.nomCategorie = nomCategorie;
    }

    // Getters and Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNomCategorie() {
        return nomCategorie;
    }

    public void setNomCategorie(String nomCategorie) {
        this.nomCategorie = nomCategorie;
    }

    public List<Cours> getCours() {
        return cours;
    }

    public void setCours(List<Cours> cours) {
        this.cours = cours;
    }

    @Override
    public String toString() {
        return nomCategorie;
    }
}